src/enumerable.cr

```diff
@@ -1172,7 +1172,14 @@ module Enumerable(T)
   # [1, 2, 3, 4].size # => 4
   # ```
   def size
-    count { true }
+    if self.is_a?(Array) || self.is_a?(Slice) ||
+       self.is_a?(Deque) || self.is_a?(BitArray) ||
+       self.is_a?(Tuple) || self.is_a?(StaticArray) ||
+       self.is_a?(Range) || self.is_a?(Set)
+      self.size
```

Comment:
If it worked, you'd called yourself recursively...
This is a no-op, since `Enumerable` is usually included at the beginning or the class, and `size` is overridden later in that class.

Reply:
I'm doing all this for methods like `Enumerable#to_a`. So the types's `size` is used there instead of `count { true }`.
Or the array creation in `to_a` becomes `ary = Array(T).new(self.size)` (which won't call this method then) but that wouldn't work for all types. Not all have `#size`.

Reply:
But... using `size` in `to_a` will call the overridden `size` method if there is one, or the 'default' one in `Enumerable` if not

Reply:
Really? Hmm I understood it the way that `self.size` will call the overridden `size` method if possible and `size` always calls `Enumerable`'s `size`. But you are probably right, I will remove these if checks now.

Reply:
Basically if you have a local variable called `size`, doing `size` will access the local var not the method, instead one can use `self.size` to call the method. There is no way to call `Enumerable`'s `size` when the includer type overrides the method `size`.